ringracers: init at 2.3

donovanglover opened the pull request:
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/306707

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

donovanglover committed:
ringracers: init at 2.3

Closes https://github.com/NixOS/nixpkgs/issues/306707

TomaSajt commented on pkgs/by-name/ri/ringracers/package.nix:
```diff
@@ -0,0 +1,110 @@
+
+  desktopItems = [
+    (makeDesktopItem rec {
+      name = "Ring Racers";
+      exec = finalAttrs.pname;
+      icon = finalAttrs.pname;
+      comment = "Kart racing mod based on SRB2";
+      desktopName = name;
+      genericName = name;
+      startupWMClass = ".ringracers-wrapped";
+      categories = [ "Game" ];
```
I think using `finalAttrs.pname` is not correct, as the actual files don't match `pname` if the `pname` is overwritten for some reason. I'd just put the `"ringracers"` literal string instead.

The `name` attibute is the name of the `.desktop` file created, so it would be `Ring Racers.desktop`. IMO having spaces in filenames should be avoided.

Not sure whether `genericName` should be set at all, if it's the same as `desktopName`.

donovanglover replied:
Fixed all

TomaSajt commented on pkgs/by-name/ri/ringracers/package.nix:
```diff
@@ -0,0 +1,110 @@
+    runHook preInstall
+
+    install -Dm644 ../srb2.png $out/share/pixmaps/ringracers.png
+    install -Dm644 ../srb2.png $out/share/icons/ringracers.png
```
Is there any reason why the icon is copied to 2 places?

`pixmaps` is fine for the `.desktop` file, but the `icons` directory usually has some extra nesting when used for `.desktop` files (e.g. `icons/hicolor/48x48/apps/foo.png`)

donovanglover replied:
Removed `pixmaps`

TomaSajt commented on pkgs/by-name/ri/ringracers/package.nix:
```diff
@@ -0,0 +1,110 @@
+  ];
+
+  cmakeFlags = [
+    "-DSRB2_ASSET_DIRECTORY=${finalAttrs.assets}/share/ringracers"
+    "-DGME_INCLUDE_DIR=${game-music-emu}/include"
+    "-DSDL2_MIXER_INCLUDE_DIR=${lib.getDev SDL2_mixer}/include/SDL2"
+    "-DSDL2_INCLUDE_DIR=${lib.getDev SDL2}/include/SDL2"
```
I've seen other packages use special `lib` functions for constructing `cmake` definitions.

For this case it would be `(lib.cmakeFeature "SRB2_ASSET_DIRECTORY" "${finalAttrs.assets}/share/ringracers")`

IMO you can keep it as is, but it's good be aware of different methods.

donovanglover replied:
Nice to know and will keep as-is

TomaSajt commented on pkgs/by-name/ri/ringracers/package.nix:
```diff
@@ -0,0 +1,110 @@
+    hash = "sha256-fiKZg/WU8PRyR8RggYHSL0rVUBeFuuQdEKyIj/O0aN4=";
+  };
+
+  assets = stdenv.mkDerivation {
+    pname = "ringracers-data";
+    version = finalAttrs.version;
+
+    src = fetchzip {
+      url = "https://github.com/KartKrewDev/RingRacers/releases/download/v${finalAttrs.version}/Dr.Robotnik.s-Ring-Racers-v${finalAttrs.version}-Assets.zip";
+      hash = "sha256-qTWDp5jfpBtEcYmzsxeE6XaohjCjSl/x8eXxnIdhCxY=";
+      stripRoot = false;
+    };
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share/ringracers
+      cp -r * $out/share/ringracers
+
+      runHook postInstall
+    '';
+  };
```
If there's a no special reason to use the `/share/ringracers` structure, then you could just use `fetchzip` by itself.
`fetchzip` already outputs a derivation that is acceptable for usage as an `assets` derivation. Though if you do this, you'll need to remove the `/share/ringracers` part from the `makeWrapper` section.

```suggestion
  assets = fetchzip {
    name = "${finalAttrs.pname}-${finalAttrs.version}-assets";
    url = "https://github.com/KartKrewDev/RingRacers/releases/download/v${finalAttrs.version}/Dr.Robotnik.s-Ring-Racers-v${finalAttrs.version}-Assets.zip";
    hash = "sha256-qTWDp5jfpBtEcYmzsxeE6XaohjCjSl/x8eXxnIdhCxY=";
    stripRoot = false;
  };
```

donovanglover replied:
Good catch

donovanglover commented:
Note: Changed `meta.homepage` to https://kartkrew.org

donovanglover commented:
Added SRB2 keyword to description for search

AAGaming00 commented:
Upstream is at 2.2 now.

donovanglover commented:
Updated thanks :+1:

TheHans255 committed:
maintainers: add TheHans255

donovanglover commented:
Things done:

- Added @TheHans255 as maintainer
- Updated desktopItem based on [flatpak upstream](https://github.com/flathub/org.kartkrew.RingRacers/blob/master/org.kartkrew.RingRacers.desktop)

TomaSajt commented on pkgs/by-name/ri/ringracers/package.nix:
```diff
@@ -0,0 +1,95 @@
+  ];
+
+  desktopItems = [
+    (makeDesktopItem rec {
+      name = "Dr. Robotnik's Ring Racers";
+      exec = "ringracers";
+      icon = "ringracers";
+      comment = "This is Racing at the Next Level";
+      desktopName = name;
+      startupWMClass = ".ringracers-wrapped";
+      categories = [ "Game" ];
+    })
```
```suggestion
    (makeDesktopItem {
      name = "ringracers";
      exec = "ringracers";
      icon = "ringracers";
      comment = "This is Racing at the Next Level";
      desktopName = "Dr. Robotnik's Ring Racers";
      startupWMClass = ".ringracers-wrapped";
      categories = [ "Game" ];
    })
```

Now that the `desktopName` is much longer I no longer think it should be included in the filename.
(The file name is just `name+".desktop"`. This also means that you can get rid of the `rec`.

santiagocezar commented:
It's at 2.3 now, hash for the source is `sha256-X2rSwZOEHtnSJBpu+Xf2vkxGUAZSNSXi6GCuGlM6jhY=` and for assets `sha256-Flfrv1vbL8NeN3sxafsHuqPPaxZMgvvohizXefUFoVg=`

donovanglover commented:
Updated thanks!

TheHans255 commented:
What else is left to do before merge? Is it just the aarch64 and Darwin builds, or are there other items?
